articles/142_idl.md

```diff
@@ -144,3 +144,95 @@ For now only a one-dimensional, fixed-size array is supported though.
+| sequence<uint8>              |                                   |                | numpy.ndarray(dtype=numpy.uint8)                |
+| sequence<uint8, N>           |                                   |                | numpy.ndarray(dtype=numpy.uint8)                |
+| int16[N]                     |                                   |                | numpy.ndarray(shape=(N, ), dtype=numpy.int16)   |
+| sequence<int16>              |                                   |                | numpy.ndarray(dtype=numpy.int16)                |
+| sequence<int16, N>           |                                   |                | numpy.ndarray(dtype=numpy.int16)                |
+| uint16[N]                    |                                   |                | numpy.ndarray(shape=(N, ), dtype=numpy.uint16)  |
+| sequence<uint16>             |                                   |                | numpy.ndarray(dtype=numpy.uint16)               |
+| sequence<uint16, N>          |                                   |                | numpy.ndarray(dtype=numpy.uint16)               |
+| int32[N]                     |                                   |                | numpy.ndarray(shape=(N, ), dtype=numpy.int32)   |
+| sequence<int32>              |                                   |                | numpy.ndarray(dtype=numpy.int32)                |
+| sequence<int32, N>           |                                   |                | numpy.ndarray(dtype=numpy.int32)                |
+| uint32[N]                    |                                   |                | numpy.ndarray(shape=(N, ), dtype=numpy.uint32)  |
+| sequence<uint32>             |                                   |                | numpy.ndarray(dtype=numpy.uint32)               |
+| sequence<uint32, N>          |                                   |                | numpy.ndarray(dtype=numpy.uint32)               |
+| int64[N]                     |                                   |                | numpy.ndarray(shape=(N, ), dtype=numpy.int64)   |
+| sequence<int64>              |                                   |                | numpy.ndarray(dtype=numpy.int64)                |
+| sequence<int64, N>           |                                   |                | numpy.ndarray(dtype=numpy.int64)                |
+| uint64[N]                    |                                   |                | numpy.ndarray(shape=(N, ), dtype=numpy.uint64)  |
+| sequence<uint64>             |                                   |                | numpy.ndarray(dtype=numpy.uint64)               |
+| sequence<uint64, N>          |                                   |                | numpy.ndarray(dtype=numpy.uint64)               |
+| array of string              | struct {size\_t, T * }, size\_t N |                |                                                 |
```

Comment:
"array" seems wrong here. The mapping suggests that it is a sequence (non static size).

Also the mapping doesn't seem to be different than the generic definition of arrays and sequences above? Should this line be removed?

Reply:
This line probably should be removed, I think it made sense when the mapping was  `rcutils_string_array`. `array of string` was trying to say `string[N]` from `T[N]` in the generic table, so the default type would be `char * [N]`.

Reply:
:heavy_check_mark: removed in 828db257ea3672c95caefa3f7d0cc2e0d932bc01.

The same commit also consolidates the `numpy` mapping into a more readable "templated" form - while the cells get wider I hope it makes it more readable / comprehensible than the verbose version before.